Commonalities/documentation/API-design-guidelines.md

```diff
@@ -1021,3 +1027,248 @@ Finally, this part describes the OAuth security applied to the API. This spec is
+| startsAt | string - datetime| Date when the subscription will begin/begun. This attribute must not be present in the `POST` request as it is provided by API server. It must be present in `GET` endpoints | optional |
+| expiresAt | string - datetime| Date when the subscription will expire. This attribute must not be present in the `POST` request as it is provided by API server.  | optional |
+| subscriptionDetail | object | Object defined for each subscription depending on the event - it could be for example the ueID targeted by the subscription | optional |
+
+The subscriptionDetail must have at least an eventType attribute:
+
+| name | type | attribute description | cardinality |
+| ----- |	-----  |	 -----  |  -----  | 
+| eventType | string | Type of event subscribed. This attribute must be present in the `POST` request. It is open to API working group to allow providing a list of event type based on specific UC | mandatory  |
+
+
+_Error definition for subscription_
+
+Error definition are described in this guideline applies for subscriptions.
+
+Following Error code must be present:
+* for `POST`: 400, 401, 403, 409, 500, 503
+* for `GET`: 400, 401, 403, 500, 503
+* for `GET/{subscriptionId}`: 400, 401, 403, 404, 500, 503
+* for `DELETE`: 400, 401, 403, 404, 500, 503
+
+_Termination for resource-based subscription_
+
+3 scenarios are possibles (business conditions may apply):
+
+* case1: subscriptionExpireTime has been provided in the request and reached. The operator in this case has to terminate the subscription.
+* case2: subscriber requests a `DELETE` for a subscription. 
+* case3: subscription ended at operator request. 
+
+It could be useful to provide a mechanism to inform subscriber for case3 (and probably case1). In this case a specific event type could be used.
+
+_Termination rules regarding subscriptionExpireTime usage_
+* When client side providing a `subscriptionExpireTime`, service side has to terminate the subscription without expecting a `DELETE` operation.
+* When the `subscriptionExpireTime` is not provided, client side has to trigger a `DELETE` operation to terminate it
+
+
+_Subscription example_
+In this example, we illustrate a request for a roaming status subscription. Requester did not provide a wished expiration time for the subscription. In the response, server accepts this request and sets a subscription end one year later. This is an illustration and each implementation is free to provide - or not - a subscription planned expiration date.
+
+Request:
+
+```
+curl -X 'POST' \
+  'http://localhost:9091//device-status/v0/subscriptions' \
+  -H 'accept: application/json' \
+  -H 'Content-Type: application/json' \
+  -d
+ ```
+ ```json 
+{
+  "notificationUrl": "https://application-server.com",
+  "notificationAuthToken": "c8974e592c2fa383d4a3960714",
```

Comment:
@eric-murray @PedroDiez @bigludo7 

token being used in `data` instead of `header`.

Reply:
This seems right for me as this token is for the POST Notification header. Here this is just passing the information.